src/transformers/models/mobilevitv2/modeling_mobilevitv2.py

```diff
@@ -0,0 +1,1073 @@
+        hidden_states = self.dropout2(hidden_states)
+        return hidden_states
+
+
+class MobileViTV2TransformerLayer(nn.Module):
+    def __init__(
+        self,
+        config: MobileViTV2Config,
+        embed_dim: int,
+        ffn_latent_dim: int,
+        dropout: float = 0.0,
+    ) -> None:
+        super().__init__()
+        # pre_norm_attn
+        # self.layernorm_before = MobileViTV2LayerNorm2D(embed_dim, eps=config.layer_norm_eps)
+        self.layernorm_before = nn.GroupNorm(num_groups=1, num_channels=embed_dim, eps=config.layer_norm_eps)
+        self.attention = MobileViTV2LinearSelfAttention(config, embed_dim)
+        self.dropout1 = nn.Dropout(p=dropout)
+        # pre_norm_ffn
+        # self.layernorm_after = MobileViTV2LayerNorm2D(embed_dim, eps=config.layer_norm_eps)
+        self.layernorm_after = nn.GroupNorm(num_groups=1, num_channels=embed_dim, eps=config.layer_norm_eps)
+        self.ffn = MobileViTV2FFN(config, embed_dim, ffn_latent_dim, config.ffn_dropout)
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        layernorm_1_out = self.layernorm_before(hidden_states)
+        attention_output = self.attention(layernorm_1_out)
+        hidden_states = attention_output + hidden_states
+
+        layer_output = self.layernorm_after(hidden_states)
+        layer_output = self.ffn(layer_output)
+
+        layer_output = layer_output + hidden_states
+        return layer_output
+
+
+class MobileViTV2Transformer(nn.Module):
+    def __init__(
+        self,
+        config: MobileViTV2Config,
+        n_layers: int,
+        d_model: int,
+    ) -> None:
+        super().__init__()
+
+        ffn_multiplier = config.ffn_multiplier
+
+        # if isinstance(ffn_multiplier, Sequence) and len(ffn_multiplier) == 2:
+        #     ffn_dims = torch.linspace(ffn_multiplier[0], ffn_multiplier[1], n_layers, dtype=float) * d_model
+        # elif isinstance(ffn_multiplier, Sequence) and len(ffn_multiplier) == 1:
+        #     ffn_dims = [ffn_multiplier[0] * d_model] * n_layers
+        # elif isinstance(ffn_multiplier, (int, float)):
+        #     ffn_dims = [ffn_multiplier * d_model] * n_layers
+        # else:
+        #     raise NotImplementedError
+
+        ffn_dims = [ffn_multiplier * d_model] * n_layers
+
+        # ensure that dims are multiple of 16
+        ffn_dims = [int((d // 16) * 16) for d in ffn_dims]
+
+        self.layer = nn.ModuleList()
+        for block_idx in range(n_layers):
+            transformer_layer = MobileViTV2TransformerLayer(
+                config, embed_dim=d_model, ffn_latent_dim=ffn_dims[block_idx]
+            )
+            self.layer.append(transformer_layer)
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        for layer_module in self.layer:
+            hidden_states = layer_module(hidden_states)
+        return hidden_states
+
+
+class MobileViTV2Layer(nn.Module):
+    """
+    MobileViTV2 block: https://arxiv.org/abs/2206.02680
```

Comment:
Perhaps we should rename this class to `MobileViTV2Block` - I'll let you decide what you think is best

Reply:
I think using `MobileViTV2Layer` is more consistent with the original MobileViT (v1) implementation.

Reply:
Sounds good!